### PR TITLE
extract helpers to simplify ensure/ensureDeleted methods

### PR DESCRIPTION
**Pull request by skriss**
Extracts some helper functions for processing the results of
"ensure" function calls to reduce duplication and improve
readability.

Signed-off-by: Steve Kriss <krisss@vmware.com>

### CODE REVIEW

File: internal/operator/controller/contour/controller.go [hunk 1]
```diff
@@ -307,62 +293,42 @@ func (r *reconciler) ensureContourForGatewayClass(ctx context.Context, contour *
...
-			r.log.Info("deleted rbac for contour", "namespace", contour.Namespace, "name", contour.Name)
+			r.log.Info(fmt.Sprintf("deleted %s for contour", resource), "namespace", contour.Namespace, "name", contour.Name)
 		}
-		if err := objns.EnsureNamespaceDeleted(ctx, cli, contour); err != nil {
-			errs = append(errs, fmt.Errorf("failed to delete namespace %s for contour %s/%s: %w",
-				contour.Spec.Namespace.Name, contour.Namespace, contour.Name, err))
+	}
+
+	switch contour.Spec.NetworkPublishing.Envoy.Type {
+	// TODO should ClusterIP be included in this list like with the create?
```
Comment (skriss) [inline internal/operator/controller/contour/controller.go:312]: cc @danehans

File: internal/operator/controller/gateway/controller.go [hunk 2]
```diff
@@ -197,149 +197,126 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
...
+	contourImage := r.config.ContourImage
+	envoyImage := r.config.EnvoyImage
+
+	handleResult("job", objjob.EnsureJob(ctx, cli, contour, contourImage))
+	handleResult("deployment", objdeploy.EnsureDeployment(ctx, cli, contour, contourImage))
+	handleResult("daemonset", objds.EnsureDaemonSet(ctx, cli, contour, contourImage, envoyImage))
+	handleResult("contour service", objsvc.EnsureContourService(ctx, cli, contour))
+
+	switch contour.Spec.NetworkPublishing.Envoy.Type {
+	// TODO ClusterIP?
```
Comment (skriss) [inline internal/operator/controller/gateway/controller.go:237]: cc @danehans

File: internal/operator/controller/gateway/controller.go [hunk 3]
```diff
@@ -197,149 +197,126 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
...
+			r.log.Info(fmt.Sprintf("deleted %s for contour", resource), "namespace", contour.Namespace, "name", contour.Name)
 		}
 	}
-	if err := objsvc.EnsureContourServiceDeleted(ctx, cli, contour); err != nil {
-		errs = append(errs, fmt.Errorf("failed to delete service for contour %s/%s: %w", contour.Namespace, contour.Name, err))
-	} else {
-		r.log.Info("deleted contour service for contour", "namespace", contour.Namespace, "name", contour.Name)
+
+	switch contour.Spec.NetworkPublishing.Envoy.Type {
+	// TODO ClusterIP?
```
Comment (skriss) [inline internal/operator/controller/gateway/controller.go:264]: cc @danehans

File: internal/operator/controller/contour/controller.go (hunk 1 above)
Comment (skriss) [reply internal/operator/controller/contour/controller.go:310]: addressed by #292

File: internal/operator/controller/gateway/controller.go (hunk 2 above)
Comment (skriss) [reply internal/operator/controller/gateway/controller.go:237]: addressed by #293

File: internal/operator/controller/gateway/controller.go (hunk 3 above)
Comment (skriss) [reply internal/operator/controller/gateway/controller.go:264]: addressed by #293

**Comment by skriss** [comment]
Gentle reminder, looking for feedback on if this approach is helpful for readability.

**Review by sunjayBhatia — APPROVED** [review]
LGTM, looks like a nice cleanup and makes the logic flow easier to read, less indents/nesting, less repetition, etc.

**Comment by skriss** [comment]
**Hold** until #293  is merged.

**Review by danehans — APPROVED** [review]
/lgtm

**Pull request merged**